clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,398 @@
+use std::{
+    fs, io,
+    path::PathBuf,
+    process::{Command, Stdio},
+    time::Duration,
+};
+
+use const_format::concatcp;
+use rust_cgi::{http::Method, text_response, Request, Response};
+use tracing::level_filters::LevelFilter;
+use uuid::Uuid;
+
+use crate::{
+    command_listener::CommandResponse,
+    config::{InterfaceConfig, MqttConfig, TeliodDaemonConfig},
+    ClientCmd, DaemonSocket, TIMEOUT_SEC,
+};
+
+const TELIOD_TMP_DIR: &str = "/tmp/nordsecuritymeshnet";
+const PID_FILE: &str = concatcp!(TELIOD_TMP_DIR, "/teliod.pid");
+const QPKG_DIR: &str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+const TELIOD_BIN: &str = concatcp!(QPKG_DIR, "/teliod");
+const MESHNET_LOG: &str = concatcp!(QPKG_DIR, "/meshnet.log");
+const TELIOD_LOG: &str = "/var/log/teliod.log";
+
+#[cfg(not(test))]
+const TELIOD_CFG: &str = concatcp!(QPKG_DIR, "/teliod.cfg");
+#[cfg(test)]
+use tests::TELIOD_CFG;
+
+pub(crate) fn handle_request(request: Request) -> Response {
+    match (request.method(), request.uri().query()) {
+        (&Method::POST, _) => start_daemon(),
+        (&Method::DELETE, _) => stop_daemon(),
+        (&Method::PATCH, _) => {
```

Comment:
verification of path?

Reply:
For example in the documentation we say:
```
- **Endpoint**: `/`
- **Method**: `POST`
```
But in code we start daemon after any post, regardless of the path. We should verify that the path is exacly `/`. Similar for `DELETE` and `PATCH` methods.

Reply:
the Endpoint path `/` is relative to the teliod.cgi path. So `/` is equivalent to `/cgi-bin/qpkg/teliod.cgi`. I can check against this hardcoded path..

Reply:
Is user able to control the path parameter?

Reply:
technically yes, but if they change the path they won't be running `teliod.cgi` 🤔

Reply:
So what is the point of checking path few lines below in:
```rust
(&Method::GET, Some("info=get-status"))
```
?

Reply:
That value is returned from `uri().query()` whereas the path is retrieved from `uri().path()`:

see [rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2fc86337eca752f3f5dd20b803789f40)

Reply:
If we expect that it will always be the same value (`/cgi-bin/qpkg/teliod.cgi`?) than there should be no reason not to check - if we are mistaken we want to know about this.

Reply:
fair enough, added

Reply:
Ugh.... I meant `query` not `path` 🤦

Reply:
Hey let's not do path verification in this PR, let's leave it after we have a web page logic
also path() should only contain string afeter cgi-bin/qpkg/teliod.cgi,soforexample if cgi-bin/qpkg/teliod.cgi/xxx/x would be request, path should be /xxx/x.

In my setup i will rely that '/' path is return index page, and '/style.css' will return css and so on, so let's leave path out of this pr.

Query tipycaly should not be fragile, we disccuseed with luis already, but proper parsing should be done so for example:
?some-id=1&some-other=2 should be equivalent to ?some-other=2&some-id=1
We will add this with auth logic

Reply:
will remove path verification. Query will indeed be validated on a next PR